packages/forklift-console-plugin/src/modules/Plans/views/details/components/MappingList.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC } from 'react';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { Button, DataList } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+
+import { Mapping, MappingListItem } from './MappingListItem';
+
+interface MappingListProps {
+  /**
+   * List of existed mappings
+   */
+  mappings: Mapping[];
+  /**
+   * List of available source provider's mappings
+   */
+  availableSources?: string[];
+  /**
+   * List of available target provider's mappings
+   */
+  availableDestinations?: string[];
+  replaceMapping?: (val: { current: Mapping; next: Mapping }) => void;
+  deleteMapping?: (mapping: Mapping) => void;
+  addMapping?: () => void;
+  generalSourcesLabel: string;
+  noSourcesLabel: string;
+  /**
+   * Is the mappings buttons and DataListCells disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Is in edit/view mode? In case of view mode, the DataListCells are disabled and buttons are hidden
+   */
+  isEditable?: boolean;
+}
+
+export const MappingList: FC<MappingListProps> = ({
+  mappings,
+  availableSources,
+  availableDestinations,
+  replaceMapping,
+  deleteMapping,
+  addMapping,
+  generalSourcesLabel,
+  noSourcesLabel,
+  isDisabled = false,
+  isEditable = true,
+}) => {
+  const { t } = useForkliftTranslation();
+
+  return (
+    <>
+      <DataList isCompact aria-label="">
+        {mappings?.map(({ source, destination }, index) => (
+          <MappingListItem
+            source={source}
+            destination={destination}
+            sources={availableSources}
+            destinations={availableDestinations}
+            replaceMapping={replaceMapping}
+            deleteMapping={deleteMapping}
+            index={index}
+            key={`${source}-${destination}`}
+            generalSourcesLabel={generalSourcesLabel}
+            noSourcesLabel={noSourcesLabel}
+            isDisabled={!isEditable || isDisabled}
+            isEditable={isEditable}
+          />
+        ))}
+      </DataList>
+      {isEditable ? (
```

Comment:
we already have `isDisabled` , no need to hide the button,
note that if you remove this line the code is identical to - https://github.com/kubev2v/forklift-console-plugin/blob/main/packages/forklift-console-plugin/src/modules/Providers/views/migrate/components/MappingList.tsx#L26

and then you can just import it, see the comment on `hooks`